# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1523127980 14400
#      Sat Apr 07 15:06:20 2018 -0400
# Node ID dd774017866bcb74e8bd9f345c6a683109f2d656
# Parent  5960fd1b56f13c85d4505970173162bade49cd6e
allow most headers to be included alone (bug #53505)

* pt-unop.h, sparse-chol.h, sparse-qr.h: Include or declare required
symbols to allow file to be included alone.
* oct.h: Don't include gripes.h.

diff --git a/libinterp/corefcn/oct.h b/libinterp/corefcn/oct.h
--- a/libinterp/corefcn/oct.h
+++ b/libinterp/corefcn/oct.h
@@ -27,17 +27,15 @@ along with Octave; see the file COPYING.
 
 // Things that are often included to create .oct files.
 
 #include "Matrix.h"
 #include "oct-locbuf.h"
 #include "defun-dld.h"
 #include "error.h"
 #include "errwarn.h"
-// gripes.h was deprecated in version 4.2 and will be removed in 4.6
-#include "gripes.h"
 #include "help.h"
 #include "ovl.h"
 #include "pager.h"
 #include "utils.h"
 #include "variables.h"
 
 #endif
diff --git a/libinterp/parse-tree/pt-unop.h b/libinterp/parse-tree/pt-unop.h
--- a/libinterp/parse-tree/pt-unop.h
+++ b/libinterp/parse-tree/pt-unop.h
@@ -25,16 +25,17 @@ along with Octave; see the file COPYING.
 
 #include "octave-config.h"
 
 #include <string>
 
 class octave_value;
 class octave_value_list;
 
+#include "ov.h"
 #include "pt-exp.h"
 #include "pt-walk.h"
 
 namespace octave
 {
   class symbol_scope;
 
   // Unary expressions.
diff --git a/liboctave/numeric/sparse-chol.h b/liboctave/numeric/sparse-chol.h
--- a/liboctave/numeric/sparse-chol.h
+++ b/liboctave/numeric/sparse-chol.h
@@ -22,16 +22,18 @@ along with Octave; see the file COPYING.
 
 */
 
 #if ! defined (octave_sparse_chol_h)
 #define octave_sparse_chol_h 1
 
 #include "octave-config.h"
 
+#include "CSparse.h"
+
 class RowVector;
 class SparseMatrix;
 class SparseComplexMatrix;
 
 namespace octave
 {
   namespace math
   {
diff --git a/liboctave/numeric/sparse-qr.h b/liboctave/numeric/sparse-qr.h
--- a/liboctave/numeric/sparse-qr.h
+++ b/liboctave/numeric/sparse-qr.h
@@ -21,16 +21,18 @@ along with Octave; see the file COPYING.
 
 */
 
 #if ! defined (octave_sparse_qr_h)
 #define octave_sparse_qr_h 1
 
 #include "octave-config.h"
 
+#include "oct-cmplx.h"
+
 class Matrix;
 class ComplexMatrix;
 class SparseComplexMatrix;
 class SparseMatrix;
 class ColumnVector;
 template <typename T> class MArray;
 
 namespace octave
